packages/rest/src/http/route.ts

```diff
@@ -161,10 +162,14 @@ export function resolvePathAndParameters(
     }
   }
 
+  const shared =
+    (getRoutePath(program, operation)?.parameters?.properties.get("shared")?.type as BooleanLiteral)
+      ?.value ?? false;
```

Comment:
@timotheeguerin this syntax feels really ugly. Is there a better way to accomplish this??

Reply:
Man, I had written helper functions for this at some point in the past but I think they got deleted in a refactoring of the ARM library.

Reply:
Well, I'm wondering if I shouldn't make `parameters: Model` but should make it an interface type instead.

Reply:
yeah it is a bit of a pain right now. Would be nice to have an helper. There is cadlTypeToJson or something similar that will convert it to a primitive object but you then should still validate the structure.

Reply:
It looks like `@route("...", {shared: "no"})`? The cast seems suspicious. This is where the decorator typing would really help so you could let the compiler take care of it. It should be an error if shared is not a boolean. Without that, something like a helper that took name, expected type, default value that can add a diagnostic on incorrect type would be useful.

Reply:
I added validation.